### DISCUSSION TITLE
VRFConsumerBaseV2Plus not found

### DISCUSSION THREAD

**Discussion by danirmz** [Cyfrin/foundry-full-course-cu#2159] [Q&A] [answered]
Hi,
when importing the  vrfcoordinator I get file not found, I already tried adding or removing src from the filepath and get error
any ideas?

<img width="839" alt="vrfcoordinator1" src="https://github.com/user-attachments/assets/105dd61b-631b-43a0-80ab-ec0ea937e5df">

<img width="825" alt="vrfcoordinator2" src="https://github.com/user-attachments/assets/8494a7b3-8420-435a-9405-d6dc117221fe">

**Comment by EngrPips** [comment, marked as answer]
Hello @vrixas, seems like you figured this out already, right?

**Reply by danirmz** [reply]
Right :)

**Reply by mlapid** [reply]
I am having the same issue, how did you manage to solve it?

**Reply by danirmz** [reply]
@mlapid  try import {VRFConsumerBaseV2Plus} from "@chainlink/contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol";

let me know if it works.

**Reply by vicjuma** [reply]
I am facing the same problem. I have worked with both versions 1.1.1 and 1.3.0. My Solidity compiler version and import statement:
```
pragma solidity ^0.8.26;

import {VRFConsumerBaseV2Plus} from "@chainlink/contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol";
```

Here is the exact error:
```
Source "lib/chainlink-brownie-contracts/contracts/src/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol" not found: File not found. Searched the following locations: "".
```
My tree file structure
```shell
tree lib/chainlink-brownie-contracts/contracts/src/v0.8/vrf/dev/ct-lottery$ 
lib/chainlink-brownie-contracts/contracts/src/v0.8/vrf/dev/
├── ArbitrumL1Fees.sol
├── BatchVRFCoordinatorV2Plus.sol
├── BlockhashStore.sol
├── interfaces
│   ├── IVRFCoordinatorV2PlusInternal.sol
│   ├── IVRFCoordinatorV2PlusMigration.sol
│   ├── IVRFCoordinatorV2Plus.sol
│   ├── IVRFMigratableConsumerV2Plus.sol
│   ├── IVRFSubscriptionV2Plus.sol
│   ├── IVRFV2PlusMigrate.sol
│   └── IVRFV2PlusWrapper.sol
├── libraries
│   └── VRFV2PlusClient.sol
├── OptimismL1Fees.sol
├── SubscriptionAPI.sol
├── testhelpers
│   ├── ExposedVRFCoordinatorV2_5_Arbitrum.sol
│   ├── ExposedVRFCoordinatorV2_5_Optimism.sol
│   ├── ExposedVRFCoordinatorV2_5.sol
│   ├── VRFConsumerV2PlusUpgradeableExample.sol
│   ├── VRFCoordinatorTestV2_5.sol
│   ├── VRFCoordinatorV2PlusUpgradedVersion.sol
│   ├── VRFCoordinatorV2Plus_V2Example.sol
│   ├── VRFMaliciousConsumerV2Plus.sol
│   ├── VRFOld.sol
│   ├── VRFV2PlusConsumerExample.sol
│   ├── VRFV2PlusExternalSubOwnerExample.sol
│   ├── VRFV2PlusLoadTestWithMetrics.sol
│   ├── VRFV2PlusMaliciousMigrator.sol
│   ├── VRFV2PlusRevertingExample.sol
│   ├── VRFV2PlusSingleConsumerExample.sol
│   ├── VRFV2PlusWrapperConsumerExample.sol
│   └── VRFV2PlusWrapperLoadTestConsumer.sol
├── TrustedBlockhashStore.sol
├── VRFConsumerBaseV2Plus.sol
├── VRFConsumerBaseV2Upgradeable.sol
├── VRFCoordinatorV2_5_Arbitrum.sol
├── VRFCoordinatorV2_5_Optimism.sol
├── VRFCoordinatorV2_5.sol
├── VRFSubscriptionBalanceMonitor.sol
├── VRFV2PlusWrapper_Arbitrum.sol
├── VRFV2PlusWrapperConsumerBase.sol
├── VRFV2PlusWrapper_Optimism.sol
└── VRFV2PlusWrapper.sol

**Reply by EngrPips** [reply]
Use `command + p` [Mac] or `cltr + p` [Windows] and input `VRFConsumerBaseV2Plus` in the search bar that will appear in `vscode` if the file truly exists in your codebase. It will bring it, and you can use the path shown when it brings it to do your import. The image below shows how what I meant.

<img width="1440" alt="Screenshot 2024-10-23 at 09 21 43" src="https://github.com/user-attachments/assets/020d6b4e-8ecb-42f7-8813-708adae89522">

**Comment by EngrPips** [comment]
Awesome
